src/transform_for_tableau.py

```diff
@@ -30,39 +42,60 @@ def combine_course_structure():
 def combine_enrollment_and_new_analytics():
 
     new_analytics =  pd.read_csv(f'{CLEANEDDATA_FOLDER}/new_analytics.csv')
-    new_analytics['user_id'] = new_analytics['global_user_id'].apply(lambda x: int(x)-112240000000000000)
-    
+    new_analytics['user_id'] = new_analytics['global_user_id'].apply(lambda x: int(x)-INST_CODE)
+    new_analytics['course_id'] = new_analytics['global_course_id'].apply(lambda x: int(x)-INST_CODE)
 
     enrollment = pd.read_csv(f'{CLEANEDDATA_FOLDER}/enrollments.csv')
 
-    # filter to active student data only
-    enrollment = enrollment.query("enrollment_type=='StudentEnrollment' & enrollment_state=='active'")[["user_id"]]
-    enrollment = enrollment.reset_index()
-    enrollment.columns = ["user_order_id", "user_id"]
-    enrollment["user_order_id"] = enrollment["user_order_id"] +1
+    try:
+        # filter to active student data only
+        
+        def get_value(x, key):
```

Comment:
If you want to avoid a try catch block, but still return None when a key isn't found, I believe you can do:
`ast.literal_eval(x).get(key, None)`

That makes the function a 1-liner so maybe a custom function definition isn't needed at all? Of course this will break if x is not a dict so if the idea is to catch non-dictionary values and return None then I'd keep as is!

Up to you based on what was intended :)

Reply:
I was having trouble implementing that in the lambda where the function is used - any suggestions?

Reply:
So turns out _yes_, we can use that suggestion from above instead...but after spending some time on it I think I'm more inclined to keep what you have lol


We could do:
```python
# original
enrollment["student"] = enrollment["user"].apply(lambda x: get_value(x, "name"))

# replacement
enrollment["student"] = enrollment["user"].apply(lambda x: ast.literal_eval(x).get("name", None)) 
```

But I think your implementation is the safer way to go. `ast.literal_eval` raises an exception if the input isn't a valid Python datatype and without the try/catch in your function, some unexpected types in those series we `.apply()` over could break the whole thing.

Not to mention the second option is uglier.